src/memoization_cached.ml

```diff
@@ -0,0 +1,27 @@
+open Stdune
+open Memoization
+open Memoization_specs
+
+module Id = Id.Make ()
+
+type 'a t = Id.t * 'a Fiber.t -> 'a Fiber.t
+
+let id_input_spec =
+  let ser x = Id.to_int x |> string_of_int in
+  let ne x y = x <> y in
+  {
+    serialize = ser;
+    print = ser;
+    not_equal = ne;
```

Comment:
yes, more precisely `let serialize = ... in let not_equal = ... in { serialize; print = serialize; not_equal}`. Otherwise I'm fine with the current version.

Reply:
I applied this in a few places where it seemed to make sense